rotkehlchen/exchanges/bitstamp.py

```diff
@@ -295,18 +315,31 @@ def query_online_deposits_withdrawals(
         # even more fun. If somehow the endpoint is not called in order then we
         # may end up with some crypto asset movements here that would need to
         # check for corresponding asset movement in the DB.
-        serialized_location = Location.BITSTAMP.serialize_for_db()
         indices_to_delete = []
         with self.db.user_write() as write_cursor:
             for idx, crypto_movement in enumerate(crypto_asset_movements):
-                write_cursor.execute(
-                    'SELECT id from asset_movements WHERE location=? AND category=? AND timestamp=? AND asset=?',  # noqa: E501
-                    (serialized_location, crypto_movement.category.serialize_for_db(), crypto_movement.timestamp, crypto_movement.asset.identifier),  # noqa: E501
+                query_result = history_db.get_history_events(
+                    cursor=write_cursor,
+                    filter_query=HistoryEventFilterQuery.make(
+                        limit=1,
+                        location=Location.BITSTAMP,
+                        to_ts=ts_ms_to_sec(crypto_movement.timestamp),
+                        event_types=[crypto_movement.event_type],
+                        event_subtypes=[crypto_movement.event_subtype],
+                        assets=(crypto_movement.asset,),
+                        order_by_rules=[('timestamp', False)],
+                    ),
+                    has_premium=True,
                 )
-                if (result := write_cursor.fetchone()) is not None:
-                    write_cursor.execute(
-                        'UPDATE asset_movements SET address=?, transaction_id=? WHERE id=?',
-                        (crypto_movement.address, crypto_movement.transaction_id, result[0]),
+                if len(query_result) != 0:
+                    matched_movement = query_result[0]
+                    extra_data = matched_movement.extra_data if matched_movement.extra_data is not None else {}  # noqa: E501
+                    assert crypto_movement.extra_data is not None  # should always be set here
+                    extra_data.update(crypto_movement.extra_data)
```

Comment:
same comments as above here. Though the direct query here looks worse than in the filter/get. the edit with the function is probably fine

Reply:
still using the edit function as suggested, but changed the assert to a type ignore.